src/string_input_popup.cpp

```diff
@@ -291,16 +300,22 @@ void string_input_popup::draw( const utf8_wrapper &ret, const utf8_wrapper &edit
             }
         }
         if( l > 0 ) {
-            mvwprintz( w, point( _startx + sx, _starty ), _underscore_color, std::string( l, '_' ) );
+            mvwprintz( w_full, point( _startx + sx, _starty ), _underscore_color, std::string( l, '_' ) );
         }
     }
     if( !edit.empty() ) {
-        mvwprintz( w, point( start_x_edit, _starty ), _cursor_color, "%s", edit.c_str() );
+        mvwprintz( w_full, point( start_x_edit, _starty ), _cursor_color, "%s", edit.c_str() );
     }
 
     // Move curses cursor to text cursor for screen readers and IME preview positioning
-    wmove( w, cursor_pos );
-    wnoutrefresh( w );
+    wmove( w_full, cursor_pos );
+    wnoutrefresh( w_full );
```

Comment:
This must be executed at the end of drawing to position the cursor correctly for screen readers and IME preview.